src/xrpld/app/tx/detail/DeleteAccount.cpp

```diff
@@ -202,19 +220,23 @@ DeleteAccount::preclaim(PreclaimContext const& ctx)
     if ((*sleDst)[sfFlags] & lsfRequireDestTag && !ctx.tx[~sfDestinationTag])
         return tecDST_TAG_NEEDED;
 
+    auto sleAccount = ctx.view.read(keylet::account(account));
```

Comment:
Check for the root account already was in the code. I just move it, without modifying, before depositPreauth check. It feel right to me because depositPreauth check do use root account and it is strange that we check it later, after usage.  
Root account check doesn't require any amendments.

Reply:
Let's consider two versions of rippled. One is an older version, which precedes this change. And one with this change. Also let's assume that both of these errors can happen at the same time. Then these two versions return different error before the amendment is enabled, right?

Reply:
No, they both return 
`"error": "srcActNotFound",  "error_code": 67,`

Reply:
reverted